aesara/tensor/basic.py

```diff
@@ -250,338 +244,88 @@ def constant(x, name=None, ndim=None, dtype=None):
 
 
-def _obj_is_wrappable_as_tensor(x):
-    try:
-        constant(x)
-        return True
-    except TypeError:
-        return False
-
-
-def numpy_scalar(data):
-    """Return a scalar stored in a numpy ndarray.
-
-    Raises
-    ------
-    NotScalarConstantError
-        If the numpy ndarray is not a scalar.
-    EmptyConstantError
-
-    """
-
-    # handle case where data is numpy.array([])
-    if data.ndim > 0 and (len(data.shape) == 0 or builtins.max(data.shape) == 0):
-        assert np.all(np.array([]) == data)
-        raise EmptyConstantError()
-    try:
-        complex(data)  # works for all numeric scalars
-        return data
-    except Exception:
-        raise NotScalarConstantError(
-            "v.data is non-numeric, non-scalar, or has more than one" " unique value",
-            data,
-        )
-
-
-_scalar_constant_value_elemwise_ops = (
-    aes.Cast,
-    aes.Switch,
-    aes.NEQ,
-    aes.EQ,
-    aes.LT,
-    aes.GT,
-    aes.LE,
-    aes.GE,
-    aes.Sub,
-    aes.Add,
-    aes.Mod,
-    aes.Mul,
-    aes.IntDiv,
-    aes.TrueDiv,
-    aes.ScalarMinimum,
-    aes.ScalarMaximum,
-)
-
-
-def get_scalar_constant_value(
-    orig_v, elemwise=True, only_process_constants=False, max_recur=10
-):
-    """Return the constant scalar(0-D) value underlying variable `v`.
-
-    If `v` is the output of dimshuffles, fills, allocs, rebroadcasts,
-    cast, OutputGuard, DeepCopyOp, ScalarFromTensor, ScalarOp, Elemwise
-    and some pattern with Subtensor, this function digs through them.
+def get_scalar_constant_value(v: Variable, only_process_constants=False, max_recur=10):
+    r"""Return the constant scalar value for `v`.
 
-    If `v` is not some view of constant scalar data, then raise a
-    NotScalarConstantError.
+    This will perform canonicalization and constant folding on a clone of `v`.
 
     Parameters
     ----------
-    elemwise : bool
-        If False, we won't try to go into elemwise. So this call is faster.
-        But we still investigate in Second Elemwise (as this is a substitute
-        for Alloc)
     only_process_constants : bool
         If True, we only attempt to obtain the value of `orig_v` if it's
         directly constant and don't try to dig through dimshuffles, fills,
         allocs, and other to figure out its value.
-    max_recur : int
-        The maximum number of recursion.
+
+
+    Raises
+    ------
+    If `v` is not some view of constant scalar data, then raise a
+    `NotScalarConstantError`.
 
     Notes
     -----
         There may be another function similar to this one in the code,
         but I'm not sure where it is.
 
     """
-    v = orig_v
-    while True:
-        if v is None:
-            # None is not a scalar (and many uses of this function seem
-            # to depend on passing it None)
+    if isinstance(v, (np.integer, int, float)):
+        return np.asarray(v)
+
+    if getattr(v, "ndim", None) != 0:
+        raise NotScalarConstantError()
```

Comment:
The current test failures seems to be originating from this particular constraint. It seems to be failing on cases like: `[1]` or `[[1]]` i.e. n-dimensional scalar values (even though there's no such thing as an n dimensional scalar :-P).

>If `v` is the output of dimshuffles, fills, allocs, rebroadcasts,
>cast, OutputGuard, DeepCopyOp, ScalarFromTensor, ScalarOp, Elemwise
>and some pattern with Subtensor, this function digs through them

Is this function still expected to take care of cases like those ?